content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx

```diff
@@ -20,26 +20,14 @@ For Prisma Client to work reliably, PgBouncer must run in [**Transaction mode**]
 
 Transaction mode offers a connection for every transaction – a requirement for the Prisma Client to work with PgBouncer.
 
-### Add `pgbouncer=true` to the connection URL
+> Note: Previous versions of the docs suggested setting `pgbouncer=true` in the connection URL. This is [no longer the case with PgBouncer `1.21.0`](https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472)
```

Comment:
I suggest keep the discarded changes in the PR and add this in as a note:

```suggestion
:::note

You no longer need to set `pgbouncer=true` if you're using [PgBouncer `1.21.0`](https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472) and above.

:::
```

Reply:
Thanks for the review, @ankur-arch! I don't think this is the right edit since "You no longer need to set" gives the impression that it doesn't hurt anything to have `pgbouncer=true` set on the connection string. However, from the linked issue ([#21531](https://github.com/prisma/prisma/issues/21531)):

>> So to recap:
>> 
>> PgBouncer <1.21.0 => Needs pgbouncer=true to work correctly
>> PgBouncer 1.21.0, max_prepared_statements = 0 => Needs pgbouncer=true to work correctly (same behavior as <1.21.0)
>> PgBouncer 1.21.0, max_prepared_statements > 0 => Will fail with > pgbouncer=true (because of https://github.com/prisma/prisma/issues/21635)
>> PgBouncer 1.21.0, max_prepared_statements > 0 => Should work fine without pgbouncer=true (still confirming)
>> PgBouncer >1.21.0 (future release) => Should work fine with and without pgbouncer=true
>> Does that match what you are seeing @​MorenoMdz?
>
> Yes, with a couple things to add.
>
> Our cloud provider has max_prepared_statements set to 100 vs the default of 0.
Having clients with different configurations, one with pgbouncer=true and others without it, will cause pgbouncer to error out on some of the prepared statements requests.

\- From https://github.com/prisma/prisma/issues/21531#issuecomment-1794661977


@JelteF also mentions (https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472):

> I released PgBouncer 1.22.0 just now, which includes the support for DEALLOCATE ALL. So with that release using pgbouncer=true won't cause breakage anymore. But `pgbouncer=true` is not needed anymore and since 1.21.0 and it is slower, so it's recommended to remove it from your connection strings.

I personally tested (albeit N of 1) `pgbouncer=true` with v1.23.1 and IIRC saw a double-digit percentage drop in query performance with a system doing 1000s of prisma transactions per second.


I can edit the PR to include the above matrix of info, but then we would need to add more of a description to the section `Set PgBouncer `max_prepared_statements` > 0` to address various different PgBouncer versions. I'm advocating for only documenting version 1.22.0  (released 2024-01-31) and newer. I imagine having docs for multiple versions/configurations of PgBouncer would add confusion for the vast majority of developers, but can edit it if you disagree.

A bigger question for you and the dev team at prisma is: 

"Should `?pgbouncer=true` be deprecated given that @JelteF, one of the maintainers of PgBouncer recommends not using it?"

Reply:
```suggestion
:::warning
We recommend not to set `pgbouncer=true` if you're using [PgBouncer `1.21.0`](https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472) and above.
:::
```